hazelcast-sql-core/src/main/java/com/hazelcast/sql/impl/calcite/HazelcastSqlToRelConverter.java

```diff
@@ -81,13 +85,27 @@ protected RexNode convertExtendedExpression(SqlNode node, Blackboard blackboard)
             return convertLiteral((SqlLiteral) node);
         } else if (node.getKind() == SqlKind.CAST) {
             return convertCast((SqlCall) node, blackboard);
+        } else if (node.getKind() == SqlKind.CASE) {
+            return convertCase((SqlCase) node, blackboard);
         } else if (node instanceof SqlCall) {
             return convertCall(node, blackboard);
         }
 
         return null;
     }
 
+    private RexNode convertCase(SqlCase sqlCase, Blackboard blackboard) {
```

Comment:
Why do we need to intercept the CASE operator conversion? What would happen if we do not do that?